acgt-tax-consultants/orchard
# Branching Round 1!

A first stab at branching. Works in it's own right, though it may take a bit of finagling to get it actually working in the context of the whole. Currently utilizes all relative paths internally, though it should be easy to add an absolute caster pass method at the end for the rest of the program if that's the direction that that part still is utilizing at the end. Handles relative symlink generation within a file run, so it already handles the idea of allowing users to see all output for that particular run (even the reused part) in the single folder.

Things that this does not do:
* Handle generic argument paths - Right now it's hardcoded to look for 'infile' and 'outfile'; with some more finagling based on whether it is a top level or dependent module. Realistically we will need to add some sort of flags (path:true? See #46) to allow for determining when an input argument is a path and when said path is static rather than hardcoding to infile/outfile.

How to test:
1) Uncomment the test code at the bottom of the file and run
2) Place text files at locations actually running the pipeline would generate them at/pull them from (i.e., in.txt in OrchardTestStruct and out.txt/out2.txt/out3.txt in the newly created OrchardTestStruct/1 directory). You don't necessarily have to do this, but all the created links will be broken if those files are missing.
3) Edit the digit parameter in the config file.
4) Rerun the test code and you should see a new '2' directory created with symlinks to all files that will be reused in that particular run present.
